File: 75 day challenge/day 01/code.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution
{
public:
    string mergeAlternately(string word1, string word2)
    {
        string mergedString;

        while (!word1.empty() and !word2.empty())
        {
            mergedString.push_back(word1.front());
            mergedString.push_back(word2.front());
            word1.erase(0, 1);
            word2.erase(0, 1);
        }

        if (!word1.empty())
        {
            mergedString += word1;
        }
        if (!word2.empty())
        {
            mergedString += word2;
        }

        return mergedString;
    }
};

int main()
{
    Solution solution;

    // Test case 1
    std::string result1 = solution.mergeAlternately("abc", "pqr");
    std::string expected1 = "apbqcr";
    std::cout << "Test Case 1: " << (result1 == expected1 ? "Passed" : "Failed") << std::endl;

    // Test case 2
    std::string result2 = solution.mergeAlternately("ab", "pqrs");
    std::string expected2 = "apbqrs";
    std::cout << "Test Case 2: " << (result2 == expected2 ? "Passed" : "Failed") << std::endl;

    // Test case 3
    std::string result3 = solution.mergeAlternately("abcd", "pq");
    std::string expected3 = "apbqcd";
    std::cout << "Test Case 3: " << (result3 == expected3 ? "Passed" : "Failed") << std::endl;

    return 0;
}

File: 75 day challenge/day 01/code.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged_string = []
        string1 = list(word1)
        string2 = list(word2)

        while string1 and string2:
            merged_string.append(string1[0])
            merged_string.append(string2[0])
            string1.pop(0)
            string2.pop(0)
        
        if string1:
            merged_string += string1
        if string2:
            merged_string += string2
        
        return ''.join(merged_string)




def main():
    solution = Solution()

    # Test case 1
    result1 = solution.mergeAlternately("abc", "pqr")
    expected1 = "apbqcr"
    print(f"Test Case 1: {'Passed' if result1 == expected1 else 'Failed'}")

    # Test case 2
    result2 = solution.mergeAlternately("ab", "pqrs")
    expected2 = "apbqrs"
    print(f"Test Case 2: {'Passed' if result2 == expected2 else 'Failed'}")

    # Test case 3
    result3 = solution.mergeAlternately("abcd", "pq")
    expected3 = "apbqcd"
    print(f"Test Case 3: {'Passed' if result3 == expected3 else 'Failed'}")

if __name__ == "__main__":
    main()
